python_modules/dagit/dagit/pipeline_run_storage.py

```diff
@@ -50,6 +52,25 @@ def __init__(self, run_id, pipeline_name, config, execution_plan):
         self.pipeline_name = pipeline_name
         self.config = config
         self.execution_plan = execution_plan
+        self._log_queue_lock = gevent.lock.Semaphore()
+        self._log_queue = []
+        self._flush_queued = False
+        self._queue_timeout = None
+
+    def _enqueue_flush_logs(self):
+        if not self._flush_queued:
+            self._queue_timeout = time.time()
+            self._flush_queued = True
+            while (time.time() - self._queue_timeout) < 1:
+                gevent.sleep(0.1)
+            with self._log_queue_lock:
+                if self._log_queue:
+                    events = self._log_queue
+                    self._log_queue = []
+
+                    for subscriber in self._subscribers:
```

Comment:
it's a bit scary to me to call into all the rx machinery etc while holding a lock. why don't we copy the log queue into the events array and then release the lock?